geomstats/geometry/connection.py

```diff
@@ -458,7 +458,90 @@ def curvature(self, tangent_vec_a, tangent_vec_b, tangent_vec_c, base_point):
         curvature : array-like, shape=[..., {dim, [n, m]}]
             Tangent vector at `base_point`.
         """
-        raise NotImplementedError("The curvature is not implemented.")
+        riemann = self.riemann_tensor(base_point)
+        curvature = gs.einsum(
+            "...ijkl, ...i, ...j, ...k -> ...l",
+            riemann,
+            tangent_vec_a,
+            tangent_vec_b,
+            tangent_vec_c,
+        )
+        return curvature
+
+    def riemann_tensor(self, base_point):
+        r"""Compute Riemannian tensor at base_point.
+
+        Parameters
+        ----------
+        base_point :  array-like, shape=[..., {dim, [n, m]}]
+            Point on the group. Optional, default is the identity.
+
+        Returns
+        -------
+        riemann_curvature : array-like, shape=[..., {dim, [n, m]}, {dim, [n, m]},
+                                                    {dim, [n, m]}, {dim, [n, m]}]
+            R_ijk^l = riemann_tensor[i,j,k,l]
+            Riemannian tensor curvature.
+        """
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        shape = base_point.shape
+        n_points = shape[0]
+        christoffels = self.christoffels(base_point)
+        jacobian_christoffels = gs.squeeze(
+            [gs.autodiff.jacobian(self.christoffels)(point) for point in base_point]
+        )
+        prod_christoffels = gs.einsum(
+            "...ijk,...klm->...ijlm", christoffels, christoffels
+        )
+        if n_points >= 2:
+            riemann_curvature = (
+                gs.transpose(jacobian_christoffels, [0, 3, 4, 2, 1])
+                - gs.transpose(jacobian_christoffels, [0, 3, 2, 4, 1])
+                + gs.transpose(prod_christoffels, [0, 4, 2, 3, 1])
+                - gs.transpose(prod_christoffels, [0, 4, 3, 2, 1])
+            )
+        else:
+            riemann_curvature = (
+                gs.transpose(jacobian_christoffels, [2, 3, 1, 0])
+                - gs.transpose(jacobian_christoffels, [2, 1, 3, 0])
+                + gs.transpose(prod_christoffels, [3, 1, 2, 0])
+                - gs.transpose(prod_christoffels, [3, 2, 1, 0])
+            )
+        return riemann_curvature
```

Comment:
I think we can avoid the if by using `gs.move_axis` if I'm not mistake and choosing axis from the end.

Reply:
Neither `gs.moveaxis` and `gs.move_axis` seem to exist as attributes of `geomstats.backend` but yes it would greatly simplify things